src/releasers/go-yoshi.ts

```diff
@@ -0,0 +1,159 @@
+import {checkpoint, CheckpointType} from '../util/checkpoint';
+import {Update} from '../updaters/update';
+import {Commit} from '../graphql-to-commits';
+
+import {ReleaseType} from 'semver';
+import * as semver from 'semver';
+
+// Generic
+import {Changelog} from '../updaters/changelog';
+
+// Commits containing a scope prefixed with an item in this array will be
+// ignored when generating a release PR for the parent module.
+const SUB_MODULES = [
+  'bigtable',
+  'bigquery',
+  'datastore',
+  'firestore',
+  'logging',
+  'pubsub',
+  'spanner',
+  'storage',
+];
+const GAPIC_PR_REGEX = /.*auto-regenerate gapics.*/;
+
+export class GoYoshi extends ReleasePR {
+  static releaserName = 'go-yoshi';
+  protected async _run() {
+    const scopeRe = /^\w+\((?<scope>.*)\):/;
+    const latestTag = await this.gh.latestTag(
+      this.monorepoTags ? `${this.packageName}-` : undefined
+    );
+    let gapicPR: Commit | undefined;
+    const commits = (
+      await this.commits({
+        sha: latestTag?.sha,
+        path: this.path,
+      })
+    ).filter(commit => {
+      // Filter commits that don't have a scope as we don't know where to put
+      // them.
+      const scope = commit.message.match(scopeRe)?.groups?.scope;
+      if (!scope) {
+        return false;
+      }
+      // Skipping commits related to sub-modules as they are not apart of the
+      // parent module.
+      for (const subModule of SUB_MODULES) {
+        if (scope.startsWith(subModule)) {
+          // TODO(codyoss): eventually gather these commits into a map so we can
+          // purpose releases for sub-modules.
+          return false;
+        }
+      }
+      // Only have a single entry of the nightly regen listed in the changelog.
+      // If there are more than one of these commits, append associated PR.
+      if (GAPIC_PR_REGEX.test(commit.message)) {
+        if (gapicPR) {
+          const issueRe = /.*(?<pr>\(.*\))$/;
+          const match = commit.message.match(issueRe);
+          if (match?.groups?.pr) {
+            gapicPR.message = `${gapicPR.message} ${match.groups.pr}`;
+          }
+          return false;
+        } else {
+          // Throw away the sha for nightly regens, will just append PR numbers.
+          commit.sha = null;
+          gapicPR = commit;
+        }
+      }
+      return true;
+    });
+
+    const cc = new ConventionalCommits({
+      commits: commits,
+      githubRepoUrl: this.repoUrl,
+      bumpMinorPreMajor: this.bumpMinorPreMajor,
+    });
+    const candidate: ReleaseCandidate = await this.coerceReleaseCandidate(
+      cc,
+      latestTag
+    );
+
+    let changelogEntry: string = await cc.generateChangelogEntry({
+      version: candidate.version,
+      currentTag: `v${candidate.version}`,
+      previousTag: candidate.previousTag,
+    });
+
+    // remove commit reference from auto-generate gapics
+    changelogEntry = changelogEntry.replace(/\s\(\[null\].*\n/, '\n');
```

Comment:
I find this logic a little bit weird, I'll take a look before I land and see if we can simplify.